src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java

```diff
@@ -30,7 +30,7 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
         }
         Throwable t = throwable.fillInStackTrace();
         filter.filter(t);
-        throw t;
+        throw throwable;
```

Comment:
Sorry it seems I was wrong. It seems that indeed your previous attempt was correct.

```java
        if (t == null) {
            return throwable;
        }
```